src/pymor/analyticalproblems/functions.py

```diff
@@ -380,3 +381,130 @@ def evaluate(self, x, mu=None):
+            interpolation_points.shape[1] == function.dim_domain
+        assert isinstance(interpolation_matrix, np.ndarray) and \
+            interpolation_matrix.shape == (len(interpolation_points),) * 2
+        assert all(isinstance(mu, Mu) for mu in snapshot_mus)
+        assert isinstance(snapshot_coefficients, np.ndarray) and \
+            snapshot_coefficients.shape == (len(interpolation_points), len(snapshot_mus))
+        assert (evaluation_points is None) == (basis_evaluations is None)
+        assert evaluation_points is None or isinstance(evaluation_points, np.ndarray) and \
+            evaluation_points.ndim == 2 and evaluation_points.shape[1] == function.dim_domain
+        assert basis_evaluations is None or isinstance(basis_evaluations, np.ndarray) and \
+            basis_evaluations.shape == (len(interpolation_points), len(evaluation_points)) + function.shape_range
+
+        self.__auto_init(locals())
+        functions = [EmpiricalInterpolatedFunctionBasisFunction(self, i) for i in range(len(interpolation_points))]
+        coefficients = [EmpiricalInterpolatedFunctionFunctional(self, i) for i in range(len(interpolation_points))]
+        super().__init__(functions, coefficients)
+        self._last_evaluation_points, self._last_basis_evaluations = evaluation_points, basis_evaluations
+        if self._last_evaluation_points is not None:
+            self._last_evaluation_points.flags.writeable = False
+            self._last_basis_evaluations.flags.writeable = False
+        self._last_mu = None
+
+    def _update_coefficients(self, mu):
+        if mu is not self._last_mu:
+            assert self.parameters.assert_compatible(mu)
+            self._last_mu = mu
+            fx = self.function.evaluate(self.interpolation_points, mu=mu)
+            if self.triangular:
+                self._last_interpolation_coefficients = solve_triangular(
+                    self.interpolation_matrix, fx, lower=True, unit_diagonal=True
+                )
+            else:
+                self._last_interpolation_coefficients = solve(self.interpolation_matrix, fx)
+
+    def _update_evaluation_points(self, x):
+        if self._last_evaluation_points is not None and np.all(x == self._last_evaluation_points):
+            return
+        if self._last_evaluation_points is not None:
+            self.logger.info('Evaluating function snapshots for new evaluation points.')
+        x.flags.writeable = False
+        self._last_evaluation_points = x
+        snapshot_evaluations = np.array([self.function.evaluate(x, mu=mu) for mu in self.snapshot_mus])
+        self._last_basis_evaluations = self.snapshot_coefficients @ snapshot_evaluations.T
+
+    def evaluate(self, x, mu=None):
+        self._update_evaluation_points(x)
+        basis_evaluations = self._last_basis_evaluations
+        interpolation_coefficients = self._evaluate_coefficients(mu)
```

Comment:
The method `self._evaluate_coefficients` does not seem to be defined anywhere.

Reply:
Ah, forgot to remove `evaluate`. Handled by `LincombFunction.evaluate`.